### ISSUE TITLE
Proposed question: 5 ways to call a JavaScript function?

### ISSUE THREAD

**Issue by whawker** [nathansmith/javascript-quiz#12]
This is a question I was once asked in a job interview. Might be appropriate here.

http://devlicio.us/blogs/sergio_pereira/archive/2009/02/09/javascript-5-ways-to-call-a-function.aspx

**Comment by dpashkevich** [comment]
+1, only I would rather call it "5 ways to set function execution context"

**Comment by davemackintosh** [comment]
I second dpashkevich on this. Good question though!

**Comment by nathansmith** [comment]
Good suggestion, I don't disagree that it's good knowledge to have.

But, I feel like aspects of that blog post are already covered…
- `apply()`
- `call()`
- using constructors
- global scope

**Comment by dpashkevich** [comment]
Right, but it's a less direct question. You can see how the person understands that all above is connected.

**Comment by nathansmith** [comment]
I'm up for adding it. Care to write a question, and an example answer?

**Comment by dpashkevich** [comment]
I kept going back and forth with the question phrasing, here's the best I've come up so far:

**Question**
Can you think of 5 different ways a `this` can be set when calling a function?

**Answer**
- 1. Global function call

``` javascript
foo();  // this === global inside the function
```
- 2. Method call

``` javascript
obj.foo();  // this === obj inside the function
```
- 3,4. `Function.call()` and `Function.apply()`

``` javascript
foo.call(obj, param1, param2, param2);  // this === obj
foo.apply(obj, [param1, param2, param2]);  // this === obj
```
- 5. Constructor call

``` javascript
new Foo();  // this === empty object with Foo's prototype
```

**Comment by john-ka-peter** [comment]
I think this within eval context is missing

**Comment by mpatel3** [comment]
+1
